app/controllers/providers/proceedings_types_controller.rb

```diff
@@ -23,8 +23,11 @@ def create
     # PATCH /provider/applications/:legal_aid_application_id/proceedings_types/:id
     def update
       authorize legal_aid_application
-      legal_aid_application.proceeding_types.clear # Remove this when multiple proceeding types required!
-      legal_aid_application.proceeding_types << proceeding_type unless legal_aid_application.proceeding_types.include?(proceeding_type)
+      ActiveRecord::Base.transaction do
+        legal_aid_application.reset_proceeding_types! # This will probably change when multiple proceeding types implemented!
+        legal_aid_application.proceeding_types << proceeding_type unless legal_aid_application.proceeding_types.include?(proceeding_type)
+        legal_aid_application.add_default_scope_limitation!
```

Comment:
shouldn't `.add_default_scope_limitation!` only be done on the next page (`used_delegated_functions_controller.rb`)?

Reply:
No.  We always add the default substatntive scope limitation when adding a proceeding type.  Then, if delegated functions are used, then we add and *additional* scope limitation for delegated functions.  I agree that it is not clear what is going on here, so I've changed it to explicitly add the *substantive* scope limitation here.